scheduler/server/stateful_scheduler.go

```diff
@@ -1176,7 +1189,9 @@ func (s *statefulScheduler) SetSchedulerStatus(maxTasks int) error {
 	if err != nil {
 		return err
 	}
-	s.config.TaskThrottle = maxTasks
+	s.setThrottle(maxTasks)
+	s.persistSettings()
```

Comment:
For all setters where we call load or persistSettings - should we interpret failure to persist as an error? Should we not update local values if the persist call fails (ensure all or nothing updated state)?

Reply:
I don't think so.... I've updated the code so persist failures return an error.  The local values will have been set, but lack of persist success just means that if the scheduler restarts it will not pick up the updated values.
so the ux is that we manually update the setting, and we should look at the response.  If we see a message that that the value was not persisted, we should retry the update again till we see that the value is persisted (or start debugging why the db write is failing).